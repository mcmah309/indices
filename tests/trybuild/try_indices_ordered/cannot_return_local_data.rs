use indices::try_indices_ordered;

fn main() {
    one();
    two();
    three();
    four();
    five();
}

fn one() -> Result<&'static mut i32, indices::TryIndicesOrderedError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (four,) = try_indices_ordered!(slice, 4)?;
    Ok(four)
}

fn two() -> Result<(&'static mut i32, &'static mut i32), indices::TryIndicesOrderedError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, four) = try_indices_ordered!(slice, 1, 4)?;
    Ok((one, four))
}

fn three() -> Result<&'static mut i32, indices::TryIndicesOrderedError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, two, four) = try_indices_ordered!(slice, 1, 2, 4)?;
    Ok(one)
}

fn four() -> Result<(&'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32), indices::TryIndicesOrderedError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, two, three, four) = try_indices_ordered!(slice, 1, 2, 3, 4)?;
    Ok((one, two, three, four))
}

fn five() -> Result<(&'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32), indices::TryIndicesOrderedError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (zero, one, two, three, four) = try_indices_ordered!(slice, 0, 1, 2, 3, 4)?;
    Ok((zero, one, two, three, four))
}
