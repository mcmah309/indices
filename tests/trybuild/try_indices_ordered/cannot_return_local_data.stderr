error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/try_indices_ordered/cannot_return_local_data.rs:11:12
   |
9  |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
10 |     let (one, _two, _three) = try_indices_ordered!(slice, 1, 2, 3).unwrap();
11 |     return one
   |            ^^^ returns a value referencing data owned by the current function
